### PR TITLE
[v7r3] DFC allows to dump more than one SE at the time

### PR DESCRIPTION
**Pull request by chaen**
BEGINRELEASENOTES
* DMS
CHANGE: DFC dump feature can dump more than one SE at the same time

ENDRELEASENOTES

### CODE REVIEW

**Review by atsareg — APPROVED** [review]

**Comment by atsareg** [comment]
Is it still WIP ?

**Comment by fstagni** [comment]
The stored procedure should be re-applied right? Then we need some instructions.

**Comment by chaen** [comment]
I'll put it in a wiki as soon as this will be merged and released, since I need a release number for that

File: src/DIRAC/Resources/Catalog/FileCatalogClient.py [hunk 1]
```diff
@@ -661,17 +664,21 @@ def getDatasetFiles(self, datasets, timeout=120):
 
     #############################################################################
 
-    def getSEDump(self, seName, outputFilename):
+    def getSEDump(self, seNames, outputFilename):
         """
-        Dump the content of an SE in the given file.
+        Dump the content of SEs in the given file.
         The file contains a list of [lfn,checksum,size] dumped as csv,
```
Comment (andresailer) [inline src/DIRAC/Resources/Catalog/FileCatalogClient.py:670]:
```suggestion
        The file contains a list of [SEName, lfn,checksum,size] dumped as csv,
```

File: src/DIRAC/DataManagementSystem/DB/FileCatalogComponents/FileManager/FileManagerBase.py [hunk 2]
```diff
@@ -1339,11 +1339,11 @@ def setFileMode(self, path, mode):
...
 
-    def getSEDump(self, seName):
+    def getSEDump(self, seNames):
         """
          Return all the files at a given SE, together with checksum and size
 
-        :param seName: name of the StorageElement
+        :param seName: list of storageElement names
 
         :returns: S_OK with list of tuples (lfn, checksum, size)
```
[the remark this answers is not in the document]
Comment (chaen) [reply src/DIRAC/DataManagementSystem/DB/FileCatalogComponents/FileManager/FileManagerBase.py:1348]: thanks

File: src/DIRAC/DataManagementSystem/DB/FileCatalogComponents/FileManager/FileManagerPs.py [hunk 3]
```diff
@@ -927,18 +927,23 @@ def _getFileLFNs(self, fileIDs):
...
 
-    def getSEDump(self, seName):
+    def getSEDump(self, seNames):
         """
          Return all the files at a given SE, together with checksum and size
 
-        :param seName: name of the StorageElement
+        :param seName: list of StorageElement names
 
         :returns: S_OK with list of tuples (lfn, checksum, size)
```
[the remark this answers is not in the document]
Comment (chaen) [reply src/DIRAC/DataManagementSystem/DB/FileCatalogComponents/FileManager/FileManagerPs.py:936]: :+1:

File: src/DIRAC/DataManagementSystem/DB/FileCatalogDB.py [hunk 4]
```diff
@@ -1177,12 +1177,12 @@ def _checkPathPermissions(self, operation, lfns, credDict):
...
-    def getSEDump(self, seName):
+    def getSEDump(self, seNames):
         """
-         Return all the files at a given SE, together with checksum and size
+         Return all the files at given SEs, together with checksum and size
 
-        :param seName: name of the StorageElement
+        :param seName: list of StorageElement names
 
         :returns: S_OK with list of tuples (lfn, checksum, size)
```
[the remark this answers is not in the document]
Comment (chaen) [reply src/DIRAC/DataManagementSystem/DB/FileCatalogDB.py:1186]: :+1:

File: src/DIRAC/DataManagementSystem/DB/FileCatalogWithFkAndPsDB.sql [hunk 5]
```diff
@@ -1941,21 +1941,26 @@ DELIMITER ;
 
 
 
--- ps_get_se_dump : dump all the lfns in an SE, with checksum and size
--- se_id : storageElement's ID
+-- ps_get_se_dump : dump all the lfns in list of SEs, with checksum and size
+-- se_id : storageElement IDs
 -- output : LFN, Checksum, Size
```
[the remark this answers is not in the document]
Comment (chaen) [reply src/DIRAC/DataManagementSystem/DB/FileCatalogWithFkAndPsDB.sql:1946]: :+1:

File: src/DIRAC/Resources/Catalog/FileCatalogClient.py [hunk 6]
```diff
@@ -89,6 +91,7 @@ class FileCatalogClient(FileCatalogClientBase):
         "deleteGroup",
         "repairCatalog",
         "rebuildDirectoryUsage",
+        "getSEDump",
```
[the remark this answers is not in the document]
Comment (chaen) [reply src/DIRAC/Resources/Catalog/FileCatalogClient.py:94]: Definitely a mistake, thanks !

File: src/DIRAC/Resources/Catalog/FileCatalogClient.py (hunk 1 above)
Comment (chaen) [reply src/DIRAC/Resources/Catalog/FileCatalogClient.py:670]: :+1:

**Review by andresailer — APPROVED** [review]

**Review by fstagni — APPROVED** [review]

**Pull request merged**